contrib/examples/walkthrough/ups-instance.yaml

```diff
@@ -7,6 +7,7 @@ spec:
   serviceClassName: user-provided-service
   planName: default
   parameters:
-    credentials:
-      name: root
-      password: letmein
+    inline:
+      credentials:
+        name: root
+        password: letmein
```

Comment:
> there's no reason not to allow for all 3 variants to appear at the same time

The reason is that supporting this is harder, and we have to come up with merging rules convention. I am not against it, but I would like to make it a separate PR once we come to an agreement. It won't be a breaking change, just a more flexible extension on top of this one.

> Also, any reason we don't allow the creds to be stored in a configMap?

No reason, just want to keep this PR simple just for the pattern of supporting different ways of representing parameters. For configMap (and any other modes) we can raise separate PRs.